src/visitor.rs

```diff
@@ -161,6 +154,13 @@ impl<'a> FmtVisitor<'a> {
             visit::FnKind::Closure => None,
         };
 
+        if let Some(ref single_line_fn) = self.rewrite_single_line_fn(&rewrite, &b) {
```

Comment:
Could we refactor so that rewrite_fn rewrites its own block rather than the visitor doing it?